src/nvim/eval.c

```diff
@@ -10929,6 +10929,89 @@ static void f_getwininfo(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   }
 }
 
+// Dummy timer callback. Used by f_wait().
+static void dummy_timer_due_cb(TimeWatcher *tw, void *data)
+{
+  if (!uv_timer_get_repeat(&tw->uv)) {
+    time_watcher_start(tw, dummy_timer_due_cb, 0, 0);
+  }
+}
+
+// Dummy timer close callback. Used by f_wait().
+static void dummy_timer_close_cb(TimeWatcher *tw, void *data)
+{
+  multiqueue_free(tw->events);
+  xfree(tw);
+}
+
+/// "wait(timeout, condition[, interval])" function
+static void f_wait(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  rettv->v_type = VAR_NUMBER;
+  rettv->vval.v_number = -1;
+
+  if (argvars[0].v_type != VAR_NUMBER) {
+    EMSG2(_(e_invarg2), "First argument of wait() must be a number");
```

Comment:
```suggestion
    EMSG2(_(e_invargval), "1");
```

Avoid highly-specific/customized error messages if the same information can be conveyed otherwise. E.g. "wait()" does not need to be mentioned because there will already be a backtrace.